### PR TITLE
Hive genesis problems

### PR DESCRIPTION
**Pull request by MarekM25**
## Changes:
- Fixed some of the problems with genesis. We're not loading accounts with balance = 0 and storage != Empty. That resulted in a genesis mismatch on hive tests. It resolves some of the tests, but not all of them.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Nethermind.State/StateProvider.cs [hunk 1]
```diff
@@ -456,7 +456,7 @@ public ChangeTrace(Account? after)
             public Account? After { get; }
         }
 
-        public void Commit(IReleaseSpec releaseSpec, IStateTracer stateTracer)
+        public void Commit(IReleaseSpec releaseSpec, IStateTracer stateTracer, bool isGenesis = false)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.State/StateProvider.cs:459]: hmmmm, I think we should discuss it with the test team and they should fix it?

Comment (tkstanczak) [reply src/Nethermind/Nethermind.State/StateProvider.cs:459]: or simply disable EIP158 in Hive during the genesis commit?

**Review by tkstanczak — CHANGES_REQUESTED** [review]
please discuss with the test team as it seems they should recreate tests or maybe we should disable EIP158 on genesis spec for Hive?

File: src/Nethermind/Nethermind.State/StateProvider.cs (hunk 1 above)
Comment (MarekM25) [reply src/Nethermind/Nethermind.State/StateProvider.cs:459]: Ok. thx. You're right. I thought about that. On the other hand, we can treat the genesis file as something that we shouldn't modify, so it has the sense to skip removing accounts from the genesis. However, I'll discuss it.

**Comment by MarekM25** [comment]
> please discuss with the test team as it seems they should recreate tests or maybe we should disable EIP158 on genesis spec for Hive?

@tkstanczak I discussed it with @fjl. To sum up: all clients apart from us don't remove such accounts from the genesis. The dust account rule should not be applied in the genesis.

File: src/Nethermind/Nethermind.State/StateProvider.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.State/StateProvider.cs:459]: I would disable through eip if possible

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**